app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -53,15 +58,15 @@ class BrowserWebViewClient @Inject constructor(private val requestRewriter: Duck
     }
 
     override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
-        webViewClientListener?.loadingStateChange(true)
+        webViewClientListener?.loadingStarted()
         webViewClientListener?.urlChanged(url)
     }
 
     override fun onPageFinished(view: WebView?, url: String?) {
-        webViewClientListener?.loadingStateChange(false)
+        webViewClientListener?.loadingFinished()
+        webViewClientListener?.urlChanged(url)
```

Comment:
why is this needed?

Reply:
The url is often different by the end of a page load

Reply:
But wouldn't you have already seen a call to `onPageStarted` whenever the URL changes?

If you have seen this happening, and the need for this call, then 👍 . but seems a bit wrong if we haven't seen that happening.